util/rtt_linux.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package util
+
+import (
+	"net"
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+func getConnRTT(conn net.Conn) (*RTTInfo, error) {
+	sysconn, ok := conn.(syscall.Conn)
+	if !ok {
+		return nil, ErrNotSyscallConn
+	}
+	raw, err := sysconn.SyscallConn()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I haven't ever worked with platform dependent stuff in Go, so maybe something makes this impossible, but can the first 10 lines or so go in the GetConnRTT wrapper?  They are identical in the platform specific cases, and probably still correct, even in the unimplemented case.